### PR TITLE
Infer preferred language from browser and send to marketo

### PR DESCRIPTION
**Pull request by petesfrench**
## Done

- Adds logic to dynamic-forms.js to get the user preferred language from their browser and attach it to the marketo payload
- Makes the function available to all forms
- Incorporates the function [here](https://github.com/canonical/ubuntu.com/blob/main/templates/base_index.html#L1342) into it
- Updates the marketo_submit function in views.py. Adds the preferredLangauge field to the enrichment_payload
- Updates the [lead enrichment](https://engage-sj.marketo.com/?munchkinId=066-EOV-335#/mktform/4198/overview/details) form to accept a preferredLangauge input
- Adds the appropriate input fields to all forms

## QA

- Fill out a form and submit it
- Check the payload has the appropriate 'preferredLangauge' field added and it matches your browsers language setting
- Check the field has been updated on the marketo lead
- Check every form had the hidden input field `preferredLanguage`. To find every form I searched for 'button type="submit"' and manually added the input field to each result that was a form.

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-1436

### CODE REVIEW

File: static/js/src/dynamic-forms.js [hunk 1]
```diff
@@ -525,6 +525,17 @@ import setupIntlTelInput from "./intlTelInput.js";
       // Setup dial code dropdown options (intlTelInput.js)
       setupIntlTelInput(phoneInput);
 
+      function setpreferredLanguage() {
+        const preferredLanguage = navigator.language || navigator.userLanguage;
+        const preferredLanguageInput = contactModal.querySelector("#preferredLanguage");
+
+        if (preferredLanguageInput) {
+          preferredLanguageInput.value = preferredLanguage || "en-US";
```
Comment (anthonydillon) [inline static/js/src/dynamic-forms.js:533]: I would default to an empty string to prevent false records

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Comment in the code

**Review by anthonydillon — CHANGES_REQUESTED** [review]
@petesfrench thank you for this huge lift. A few issues I have noticed are:
1. When using the dynamic form for example https://ubuntu-com-12692.demos.haus/internet-of-things#get-in-touch. There is duplicated `preferredLanguage` id.
2. The dynamic-forms js only runs on the modal pages. Not on static forms such as https://ubuntu-com-12692.demos.haus/internet-of-things/contact-us-hannover.

Due to this, I don't seem to be able to QA the submission of the new field. I will QA once the above comments are addressed.

**Comment by petesfrench** [comment]
Thanks for the review @anthonydillon! I have now included the function in static forms also and refactored it to encompass the preferred language check that was made in base_index.html.

I couldn't see what you mean by duplicate ID's though?

**Comment by anthonydillon** [comment]
Thanks Pete, you need to search with the inspector to see it. The page in the comment above has a newsletter sign-up form at the bottom and when you open the dynamic form you end up with two fields with the id of preferredLanguage. This means the script is only updating the value of the newsletter form not the dynamic form one.

**Comment by petesfrench** [comment]
I see now, I believe this should be fixed in my most recent commit. Can you take another look please @anthonydillon ?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Code looks good but when QA'ing I get this error from the backend in [Sentry](https://sentry.is.canonical.com/canonical/ubuntu-com/issues/21129/?query=is%3Aunresolved): 
```
'message': 'Field 'preferredLanguage' not found in form definition'
```
Each field we send has to be a field on the forms to successfully submit. Therefore this will cause all forms without this field to error.

**Comment by petesfrench** [comment]
I have removed 'preferredLanguage' from the regular form payload, so now it is only delivered in the enrichment payload. This should stop this error from occuring.

**Comment by mtruj013** [comment]
@anthonydillon could you take another look please?

**Review by anthonydillon — APPROVED** [review]
LGTM :+1: thanks for the fixes @petesfrench

**Pull request merged**